ledger/participant-integration-api/src/main/scala/platform/store/backend/h2/H2StorageBackend.scala

```diff
@@ -132,6 +97,24 @@ private[backend] object H2StorageBackend
       .as(get[Long](1).singleOpt)(connection)
   }
 
+  object H2QueryStrategy extends QueryStrategy {
+
+    override def arrayIntersectionNonEmptyClause(
+        columnName: String,
+        parties: Set[Party],
+        paramNamePostfix: String,
+    ): (String, List[NamedParameter]) =
+      (
+        if (parties.isEmpty)
+          "false"
+        else
+          parties.view.map(p => s"array_contains($columnName, '$p')").mkString("(", " or ", ")"),
```

Comment:
I think it could, H2 has an array_length function, so we can build here an sql query ensuring non-empty intersection of those arrays. Not sure it will be less or more efficient since the array_intersection is a scala code binding to H2.
It was not my goal to improve much on the H2 part of the query building. For example: in Oracle and Postgres I also put more emphasize to pass arguments as data instead of SQL code as well -- if that is the motivation behind the question.
If needed I would be ok to improve on the H2 part here in this PR on in a separate one as well.

Reply:
I was mainly asking because we use `array_intersection` in `H2EventStrategy.filteredEventWitnessesClause` a few lines below. No need to change anything in this PR, especially since it's not clear whether it's better.